src/lib/falcor/FalcorController.js

```diff
@@ -34,41 +34,76 @@ export default class FalcorController extends BaseComponent {
   // this should always contain all the data the component needs
   loadFalcorCache() {
     console.log("LOADING FROM CACHE")
-    const falcorPath = this.constructor.getFalcorPath(this.props.params)
-
-    const data = this.props.model.getCache(falcorPath)
+    let falcorPath = this.constructor.getFalcorPath(this.props.params)
+    // If the component doesn't want any data
+    if (!falcorPath || falcorPath.length === 0) {
```

Comment:
All this fixing code, (checking if length == 0), etc is repeated. We should factor it out into its own function, like validateFalcorPath() or something.